src/libraries/System.Globalization/tests/CultureInfo/CultureInfoNames.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.RemoteExecutor;
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CultureInfoNames
+    {
+        private static bool IsIcuAndRemoteExecutionSupported => RemoteExecutor.IsSupported && PlatformDetection.IsIcuGlobalization;
+        [ConditionalTheory(nameof(IsIcuAndRemoteExecutionSupported))]
+        [InlineData("en", "en", "English", "English")]
+        [InlineData("en", "fr", "English", "anglais")]
+        [InlineData("aa", "aa", "Afar", "Afar")]
+        [InlineData("en-US", "en-US", "English (United States)", "English (United States)")]
+        [InlineData("en-US", "fr-FR", "English (United States)", "anglais (États-Unis)")]
+        [InlineData("en-US", "de-DE", "English (United States)", "Englisch (Vereinigte Staaten)")]
+        [InlineData("aa-ER", "aa-ER", "Afar (Eritrea)", "Afar (Eritrea)")]
+        [InlineData("", "en-US", "Invariant Language (Invariant Country)", "Invariant Language (Invariant Country)")]
+        [InlineData("", "fr-FR", "Invariant Language (Invariant Country)", "Invariant Language (Invariant Country)")]
+        [InlineData("", "", "Invariant Language (Invariant Country)", "Invariant Language (Invariant Country)")]
+        public void TestDisplayName(string cultureName, string uiCultureName, string nativeName, string displayName)
+        {
+            RemoteExecutor.Invoke((locale, uiLocale, native, display) =>
+            {
+                CultureInfo ci = CultureInfo.GetCultureInfo(locale);
+                CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo(uiLocale);
+
+                Assert.Equal(native, ci.NativeName);
+                Assert.Equal(display, ci.DisplayName);
```

Comment:
Do you want to add a test that:

1. Sets the CurrentUICulture to a culture
2. Verifies `ci.DisplayName` is correct
3. Sets the CurrentUICulture to a different culture
4. Verifies `ci.DisplayName` is correct for the new culture

?

Reply:
We already have tests validating setting the CurrentUICulture:

https://github.com/dotnet/runtime/blob/190c69d78f42f6d312d66e3676eef3a714f7bdf2/src/libraries/System.Globalization/tests/CultureInfo/CultureInfoCurrentCulture.cs#L43

https://github.com/dotnet/runtime/blob/190c69d78f42f6d312d66e3676eef3a714f7bdf2/src/libraries/System.Globalization/tests/CultureInfo/CultureInfoAsync.cs#L17

Reply:
Those tests are obviously not testing the scenario I described above, or else they aren't running.

From your description of this change:

> Display names were cached which is wrong. The display names should be treated as the localized resources and should be retrieved every time using the CurrentUICulture.

I'm asking for a test of this functionality.

Reply:
I was trying to say, Current UI Culture wouldn't be cached if you reset it and we are already testing that. It doesn't matter the scenario you are using it for (getting resources, display names...etc.). Adding the suggested test is just testing the UI culture is not cached. I'll add the test if you think this adding any value.

Reply:
> Adding the suggested test is just testing the UI culture is not cached.

This is not true. The suggested test is testing the `ci.DisplayName` is not cached. It is to test the following change being made, from the description above:

> Display names were cached which is wrong. The display names should be treated as the localized resources and should be retrieved every time using the CurrentUICulture.